### PR TITLE
Moved examples into framework-specific subfolders

### PR DESCRIPTION
**Pull request by tgaddair**

### CODE REVIEW

**Comment by tgaddair** [comment]
@EnricoMi what do you think about this organization for examples?

File: .gitignore [hunk 1]
```diff
@@ -14,9 +14,9 @@ docs/_build
 env
 venv/
 
-examples/checkpoint*
+examples/**/checkpoint*
+examples/**/checkpoints/
```
Comment (EnricoMi) [inline .gitignore:18]: doesn't `examples/**/checkpoint*` match `examples/**/checkpoints/?

**Review by EnricoMi — COMMENTED** [review]
I like the re-org. I think under `spark` and `elastic` we should see similar framework subfolders for each of the ML frameworks.

**Comment by EnricoMi** [comment]
>    619 tests +     10     575 heavy_check_mark +     10       44 zzz ±    0  0 heavy_multiplication_x ±0 

I am wondering, why there are 10 new tests. I will look into the artifacts for that.

**Comment by EnricoMi** [comment]
> > 619 tests +     10     575 heavy_check_mark +     10       44 zzz ±    0  0 heavy_multiplication_x ±0
> 
> I am wondering, why there are 10 new tests. I will look into the artifacts for that.

Reason is that the reference master build failed (cancelled): https://buildkite.com/horovod/horovod/builds/3951 So some test did not run in the base reference.

File: .gitignore (hunk 1 above)
Comment (tgaddair) [reply .gitignore:18]: Yes, it should, good point.  That was a holdover from the previous version.  Will fix that.

**Comment by tgaddair** [comment]
Thanks for the review, @EnricoMi!  Added additional subdirectories under `spark` and `elastic`.

**Review by EnricoMi — APPROVED** [review]
I love this reorg!

**Comment by EnricoMi** [comment]
Heads up: this will conflict with #2291

**Comment by tgaddair** [comment]
Good point @EnricoMi.

cc @richardliaw heads up that the Ray examples will need to move into `examples/ray` when you rebase the Elastic Ray branch.

**Comment by richardliaw** [comment]
got it, thanks for the heads up!

**Pull request merged**